runtime/moonbeam/tests/common/mod.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+#![allow(dead_code)]
```

Comment:
can we put this over specific dead code blocks instead of over the entire file, just so we can see where the dead code is? See `with_inflation` method on `ExtBuilder` for putting it over a specific method.

I know this could have gotten in earlier and I might've just noticed it now, but I'd still make the same comment.

Reply:
That's way more boilerplatey but I guess that's ok. Will change.

Reply:
Thanks, it shows us all the code that isn't being actively used. 

I'd prefer to just take out all dead code, but there are have been suggestions in the past to add `with_inflation` for example. Still, I'd prefer to only include code when we're using it.

Reply:
This doesn't block merging, I can always remove it and compile to see the dead code whenever I'm curious.

Reply:
It's of no use to remove the annotation, we will get warning for everything in `common` no matter what.

The reason for that is although we are using the submodule in the integration test files (`use common::*`), the `common` module itself will be also compiled as an independent crate (still lives in the _special_ tests folder and gets that treatment),  throwing some unused warnings along the way.

The important thing to notice is that even if there is _real_ unused code on that file (we forgot to remove something or whatever), is not part of moonbeam itself, is part of the isolated `common` crate compiled on the fly to run the tests, so IMO it is safe (and justified) to just annotate the whole file.

Anyway I agree this _smells_. Do you know if there is any other way of defining a common submodule for integration tests and get valid linting @nanocryk ?